jolt-core/src/zkvm/ram/mod.rs

```diff
@@ -577,20 +577,7 @@ where
     ) -> Vec<Box<dyn SumcheckInstance<F, ProofTranscript>>> {
         // Accumulate advice polynomials if present
         prover_accumulate_advice(state_manager);
-
-        let val_evaluation = ValEvaluationSumcheck::new_prover(
-            self.initial_memory_state.as_ref().unwrap(),
-            state_manager,
-        );
-        let val_final_evaluation = ValFinalSumcheck::new_prover(state_manager);
-
-        #[cfg(feature = "allocative")]
-        {
-            print_data_structure_heap_usage("RAM ValEvaluationSumcheck", &val_evaluation);
-            print_data_structure_heap_usage("RAM ValFinalSumcheck", &val_final_evaluation);
-        }
-
-        vec![Box::new(val_evaluation), Box::new(val_final_evaluation)]
+        vec![]
```

Comment:
same here, I think we can move the `prover_accumulate_advice` and `verifier_accumulate_advice` to stage 4 and then delete the `stage3_prover_instances` and `stage3_verifier_instances` implementations